File: python/test/test_player.py
import pytest

from othello.board import Board
from othello.player import Player
from othello.utils import Disk


@pytest.fixture
def white_human():
    return Player(Disk.WHITE)


@pytest.fixture
def black_human():
    return Player(Disk.BLACK)


@pytest.fixture
def white_computer():
    return Player(Disk.WHITE, human=False)


@pytest.fixture
def black_computer():
    return Player(Disk.BLACK, human=False)


@pytest.fixture
def board_start_8():
    return Board(8)

File: csharp/Othello.Tests/TestPlayer.cs
namespace Othello.Tests;

public class TestPlayer {
    [Fact]
    public void NewPlayer() {
        Player player = new(Disk.Black, PlayerSettings.Default());
        Assert.Equal(Disk.Black, player.Disk);
        Assert.True(player.Human());
        Assert.True(player.CanPlay);
        Assert.Equal(0, player.RoundsPlayed);
        Assert.Equal(PlayerSettings.Default(), player.Settings);
    }

    [Fact]
    public void ResetPlayer() {
        Player player = new(Disk.White, PlayerSettings.Default());
        player.CanPlay = false;
        player.RoundsPlayed = 10;
        player.Reset();
        Assert.True(player.CanPlay);
        Assert.Equal(0, player.RoundsPlayed);
    }

    [Fact]
    public void SetHuman() {
        Player player = new(Disk.Black, PlayerSettings.Default());
        player.SetComputer();
        Assert.True(player.Computer());
        Assert.False(player.Human());

        player.SetHuman();
        Assert.True(player.Human());
        Assert.False(player.Computer());
    }

    [Fact]
    public void PlayerTypeString() {
        Player player = new(Disk.Black, PlayerSettings.Default());
        Assert.Equal("Human   ", player.TypeString());

        player.SetComputer();
        Assert.Equal("Computer", player.TypeString());
    }
}
